pallets/liquidity-rewards/src/lib.rs

```diff
@@ -299,9 +226,9 @@ pub mod pallet {
 						epoch_data.reward = changes.reward.unwrap_or(epoch_data.reward);
 						epoch_data.duration = changes.duration.unwrap_or(epoch_data.duration);
 
-						let ends_on = ends_on.ensure_add(epoch_data.duration)?;
+						let ends_on = ends_on.max(current_block).ensure_add(epoch_data.duration)?;
```

Comment:
Because now, the initial `ends_on` not uses the current block number, we need to ensure our last `ends_on` is not too old. Suppose the following scenario:
- Block 1000 => Upgrade chain with `pallet-liquidiy-rewards`, initial epoch duration: 20
- Block 1001 => `on_initialize()` is called, next ends_on is 20 + 20 = 40 (**but we wanted 1021**)
- Block 1002 => `on_initialize()` is called, next ends_on is 40 + 20 = 60
- And so on until we reach the current chain block number